pkg/ebpftracer/c/tracee.bpf.c

```diff
@@ -6735,6 +6735,338 @@ int cgroup_rmdir_signal(struct bpf_raw_tracepoint_args *ctx)
     return 0;
 }
 
+// Processes Lifecycle
+
+// NOTE: sched_process_fork is called by kernel_clone(), which is executed during
+//       clone() calls as well, not only fork(). This means that sched_process_fork()
+//       is also able to pick the creation of LWPs through clone().
+
+SEC("raw_tracepoint/sched_process_fork")
+int sched_process_fork_signal(struct bpf_raw_tracepoint_args *ctx)
+{
+    controlplane_signal_t *signal = init_controlplane_signal();
+    if (unlikely(signal == NULL))
+        return 0;
+
+    struct task_struct *parent = (struct task_struct *) ctx->args[0];
+    struct task_struct *child = (struct task_struct *) ctx->args[1];
+    struct task_struct *leader = get_leader_task(child);
+    struct task_struct *up_parent = get_leader_task(get_parent_task(leader));
```

Comment:
I understand parent and child, but what is leader and up_parent?

Reply:
The leader is the first thread in a process (meaning it is the thread group leader).The `up_parent` is then the real parent process of this process, as the parent of a thread could be another thread 😅 

This article gives a nice overview 
https://myaut.github.io/dtrace-stap-book/kernel/proc.html